dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -203,13 +204,14 @@ public async Task<SKContext> RunAsync(ContextVariables variables, CancellationTo
                     return context;
                 }
             }
-            catch (Exception e) when (!e.IsCriticalException())
+            catch (Exception e)
```

Comment:
By removing the function `IsCriticalException` we are pratically considering any exceptions to be "critical" (should throw).

Two options: 
1. Remove all the catch block
2. Keep the catch just to log and `throw` after